src/Redux/Missions/mission.js

```diff
@@ -2,22 +2,59 @@ import { createAsyncThunk } from '@reduxjs/toolkit';
 
 const GET_MISSIONS = 'space-hub/missions/GET_MISSIONS';
 
-const getMissions = createAsyncThunk(GET_MISSIONS, async (args, { dispatch }) => {
-  const response = await fetch('https://api.spacexdata.com/v3/missions');
-  const missions = await response.json();
-  dispatch({
-    type: GET_MISSIONS,
-    payload: missions,
-  });
+const JOINED_MISSION = 'space-hub/missions/JOINED_MISSION';
+const LEFT_MISSION = 'space-hub/missions/LEFT_MISSION';
+
+const getMissions = createAsyncThunk(
+  GET_MISSIONS,
+  async (args, { dispatch }) => {
+    const response = await fetch('https://api.spacexdata.com/v3/missions');
+    const missions = await response.json();
+    dispatch({
+      type: GET_MISSIONS,
+      payload: missions,
+    });
+  },
+);
+
+const joinMission = (mission) => ({
+  type: JOINED_MISSION,
+  payload: mission,
 });
 
-export { getMissions };
+const leaveMission = (mission) => ({
+  type: LEFT_MISSION,
+  payload: mission,
+});
+
+export { getMissions, joinMission, leaveMission };
 const initialState = [];
 
 const missionsReducer = (state = initialState, action) => {
   switch (action.type) {
     case GET_MISSIONS:
       return [...action.payload];
+    case JOINED_MISSION:
+      return state.map((mission) => {
+        if (mission.mission_id === action.payload) {
+          return {
+            ...mission,
+            reserved: true,
+          };
+        }
+        return mission;
+      });
+    case LEFT_MISSION:
+      return state.map((mission) => {
+        if (mission.mission_id === action.payload) {
+          return {
+            ...mission,
+            reserved: false,
```

Comment:
- [ ]  **Please Use the Reducer Available in ``` profile.js ``` and Define Your Cases There.**

**The goal is to concat all your missions in the global state's profile property. (i.e ```state.profile.joined```)** 

![State](https://user-images.githubusercontent.com/104627775/192882618-7647af5c-a641-4a48-9f9b-9c0bf2e73bd2.PNG)


 **You can accomplish this by defining your action creators in ``` profile.js ``` and then adding your cases in the ``` profileReducer () ``` , This will also allow the persist Library and method to preserve the global state on the user's Refresh**